fv3core/stencils/divergence_damping.py

```diff
@@ -1,58 +1,84 @@
-
-
-@gtstencil()
-def vorticity_main(v: sd, ua: sd, cosa_u: sd, sina_u: sd, dxc: sd, vort: sd):
-    with computation(PARALLEL), interval(...):
-        vort[0, 0, 0] = (v - 0.5 * (ua[-1, 0, 0] + ua) * cosa_u) * dxc * sina_u
+from fv3core.utils.typing import FloatField
 
 
 @gtstencil()
-def vorticity_x_edge(v: sd, uc: sd, dxc: sd, sin_sg3: sd, sin_sg1: sd, vort: sd):
-    with computation(PARALLEL), interval(...):
-        vort[0, 0, 0] = v * dxc * sin_sg3[-1, 0, 0] if uc > 0 else v * dxc * sin_sg1
-
+def damping_nord0(
+    u: FloatField,
+    v: FloatField,
+    ua: FloatField,
+    va: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+    cosa_u: FloatField,
+    cosa_v: FloatField,
+    sina_u: FloatField,
+    sina_v: FloatField,
+    dxc: FloatField,
+    dyc: FloatField,
+    sin_sg1: FloatField,
+    sin_sg2: FloatField,
+    sin_sg3: FloatField,
+    sin_sg4: FloatField,
+    rarea_c: FloatField,
+    ptc: FloatField,
+    vort: FloatField,
+    delpc: FloatField,
+    ke: FloatField,
+    da_min_c: float,
+    d2_bg: float,
+    dt: float,
+):
+    from __externals__ import i_end, i_start, j_end, j_start, namelist
 
-@gtstencil()
-def delpc_main(vort: sd, ptc: sd, delpc: sd):
     with computation(PARALLEL), interval(...):
-        delpc[0, 0, 0] = vort[0, -1, 0] - vort + ptc[-1, 0, 0] - ptc
+        ptc = (u - 0.5 * (va[0, -1, 0] + va) * cosa_v) * dyc * sina_v
+        with horizontal(region[:, j_start], region[:, j_end + 1]):
+            ptc = u * dyc * sin_sg4[0, -1, 0] if vc > 0 else u * dyc * sin_sg2
+        vort_copy = vort
+        vort = (v - 0.5 * (ua[-1, 0, 0] + ua) * cosa_u) * dxc * sina_u
+        with horizontal(region[i_start, :], region[i_end + 1, :]):
+            vort = vort_copy
+        with horizontal(region[i_start, :], region[i_end + 1, :]):
+            vort = v * dxc * sin_sg3[-1, 0, 0] if uc > 0 else v * dxc * sin_sg1
+        delpc = vort[0, -1, 0] - vort + ptc[-1, 0, 0] - ptc
+        with horizontal(region[i_start, j_start], region[i_end + 1, j_start]):
+            delpc = remove_extra_term_south_corner(vort, delpc)
+        with horizontal(region[i_start, j_end + 1], region[i_end + 1, j_end + 1]):
+            delpc = remove_extra_term_north_corner(vort, delpc)
+        delpc = rarea_c * delpc
+        delpcdt = delpc * dt
+        absdelpcdt = delpcdt if delpcdt >= 0 else -delpcdt
+        damp = damp_tmp(absdelpcdt, da_min_c, d2_bg, namelist.dddmp)
+        vort = damp * delpc
+        ke = ke + vort
 
 
-@gtstencil()
-def corner_south_remove_extra_term(vort: sd, delpc: sd):
-    with computation(PARALLEL), interval(...):
-        delpc[0, 0, 0] = delpc - vort[0, -1, 0]
+@gtscript.function
+def remove_extra_term_south_corner(extra: FloatField, field: FloatField):
+    # from __externals__ import i_start, i_end,  j_start
+    # TODO: why does this not work?
+    # with horizontal(region[i_start, j_start], region[i_end + 1, j_start]):
+    #    field = field - extra[0, -1, 0]
+    # return field
```

Comment:
I don't know, but I'd also like an answer. I guess we just have to do regions outside of function calls?